src/lib.rs

```diff
@@ -96,6 +134,46 @@ mod unix {
             metadata.is_file() && permissions.mode() & 0o111 != 0
         }
     }
+
+    /// Could this target path be ran with executable permissions
+    ///  by this runtime's run-level user?
+    ///
+    ///  Suppose the GID for the file in question is (Gm).
+    ///  Assuming that the set of all groups shared by the user, 
+    ///  (G* := { Gn, Gn+1, Gn+2, ...}), is a superset of the set whose only
+    ///  entry is the user's GID, (G_user := G* - Gn).
+    ///  Then, Ǝ a possibility some arbitary GID, like (Gn+14) for example,
+    ///  happens to be the GID belonging to group on that file.
+    ///  
+    ///  In other words, we'll collect each of the (G*) entries
+    ///  and see if there's a match. Otherwise, we check who owns the file and
+    ///  perform a similar check.
+    impl IsPermitted for Path {
```

Comment:
The formatting within this `impl` is questionable. I'd recommend trying to use `rustfmt` if you can do without massively changing other parts of the lib, or tidying this up yourself by hand